app/move/controllers/unassign.test.js

```diff
@@ -0,0 +1,263 @@
+
+const controller = new UnassignController({ route: '/' })
+
+describe('Move controllers', function() {
+  describe('Unassign controller', function() {
+    describe('#middlewareChecks()', function() {
+      beforeEach(function() {
+        sinon.stub(FormWizardController.prototype, 'middlewareChecks')
+        sinon.stub(controller, 'use')
+
+        controller.middlewareChecks()
+      })
+
+      it('should call parent method', function() {
+        expect(FormWizardController.prototype.middlewareChecks).to.have.been
+          .calledOnce
+      })
+
+      it('should call checkAllocation middleware', function() {
+        expect(controller.use).to.have.been.calledOnceWithExactly(
+          controller.checkAllocation
+        )
+      })
+    })
+
+    describe('#middlewareLocals()', function() {
+      beforeEach(function() {
+        sinon.stub(FormWizardController.prototype, 'middlewareLocals')
+        sinon.stub(controller, 'use')
+
+        controller.middlewareLocals()
+      })
+
+      it('should call parent method', function() {
+        expect(FormWizardController.prototype.middlewareLocals).to.have.been
+          .calledOnce
+      })
+
+      it('should call setMoveRelationships middleware', function() {
+        expect(controller.use.firstCall).to.have.been.calledWith(
+          controller.setMoveRelationships
+        )
+      })
+
+      it('should call correct number of middleware', function() {
+        expect(controller.use.callCount).to.equal(1)
+      })
+    })
+
+    describe('#checkAllocation()', function() {
+      let res, next
+
+      beforeEach(function() {
+        next = sinon.stub()
+      })
+
+      context('with no allocation', function() {
+        beforeEach(function() {
+          res = {
+            locals: {
+              move: {
+                id: '12345',
+              },
+            },
+            redirect: sinon.stub(),
+          }
+          controller.checkAllocation({}, res, next)
+        })
+
+        it('should redirect to allocation', function() {
+          expect(res.redirect).to.be.calledOnceWithExactly('/move/12345')
+        })
+
+        it('should not call next', function() {
+          expect(next).to.not.be.called
+        })
+      })
+
+      context('with allocation but not person', function() {
+        beforeEach(function() {
+          res = {
+            locals: {
+              move: {
+                id: '12345',
+                allocation: { id: '6789' },
+              },
+            },
+            redirect: sinon.stub(),
+          }
+          controller.checkAllocation({}, res, next)
+        })
+
+        it('should redirect to allocation', function() {
+          expect(res.redirect).to.be.calledOnceWithExactly('/allocation/6789')
+        })
+
+        it('should not call next', function() {
+          expect(next).to.not.be.called
+        })
+      })
+
+      context('with allocation and person', async function() {
```

Comment:
You can also remove the `async` here as it's no longer an `await` on the function call.

Reply:
👍